modules/040-node-manager/hooks/get_crds.go

```diff
@@ -444,6 +461,177 @@ func getCRDsHandler(input *go_hook.HookInput) error {
 	return nil
 }
 
+type capacity struct {
+	CPU    int `json:"cpu,omitempty"`
+	Memory int `json:"memory,omitempty"`
+}
+
+func calculateNodeCapacity(instanceClass string, instanceSpec interface{}) *capacity {
+	switch instanceClass {
+	case "VsphereInstanceClass":
+		s := struct {
+			CPU    int `json:"numCPUs"`
+			Memory int `json:"memory"`
+		}{}
+
+		data, _ := json.Marshal(instanceSpec)
```

Comment:
NIT: Can we avoid marshaling/unmarshalling here? Or at least do not copy it for each case.

Reply:
InstanceClass spec it an `interface{}` which in fact contains `map[string]interface{}`. I decided, that marshalling/unmarshalling of a couple of fields would be better then type casting. Especially for nested structs like `resourcesSpec` for Yandex provider.

I will try to simplify this function

Reply:
Done